### ISSUE TITLE
[CC] Item inserter resets ME terminals

### ISSUE THREAD

**Issue by Kobra331** [ReikaKalseki/Reika_Mods_Issues#1813]
Hi,
in my base the item inserter resets the text field of nearby ME terminals if it inserts items from a Chest into the Extractor. The Terminal and the Inserter are not in the same Chunk AFAIK. Other Terminals are not affected.

**Comment by ReikaKalseki** [comment]
I doubt I can change this.

**Comment by ReikaKalseki** [comment]
I realized I misread this the first time; you are saying the inserter is causing ME terminals that it does not interact with to reset? That makes no sense whatsoever, and sounds like the terminals are losing data or something.

**Comment by Kobra331** [comment]
yep thats right. Its really annoying. For example: You search Iron in the Terminal but after typing I and R the inserter inserting something in the extractor, the terminal resets and searches only for "on".
The only connection to the ME system is an export bus to the enderchest which fills via item conduit the Inserter.

**Comment by ReikaKalseki** [comment]
Are you sure it is not just lag? Oftentimes a GUI will open clientside first, allow you to input some data, and then open serverside later, refreshing the clientside one and clearing whatever data it had. I have also had this happen with ME terminals.

**Comment by Kobra331** [comment]
The terminal works fine if i pump the items directy into the extractor without inserter.
Terminals which are farer away works fine too.

**Comment by ReikaKalseki** [comment]
Show me the setup and the inserter GUI.

**Comment by Kobra331** [comment]
sure :)
https://imgur.com/a/uzfgi26
Start with the last picture and ends with the first. I uploaded them in the wrong order.

I hope this will help you :)

**Comment by ReikaKalseki** [comment]
Honestly, I have no idea. They are indeed in different chunks - meaning anything like clearing chunk data is out - and there is nothing special about the item inserter's method of adding items to an inventory when in inventory mode (as yours is).

**Comment by OvermindDL1** [comment]
Makes me think the incoming area packet interferes.  Wonder if one or the other is not registering something right or if the ME Interface GUI is listening for the wrong custom packet or so...  It would be client-side only and that does seem to match...  Would need to hook up a debugger to know for sure.  Regardless following the packet of the item inserter update to the client in a debugger and seeing what responds to it would be the thing to test with if @Kobra331 can do so?

**Comment by Kobra331** [comment]
I would like to help you, but i need some instruction. Which debugger? The Dragon API debugger or another programm/mod? And how can i follow the packet?

**Comment by OvermindDL1** [comment]
Dragon API has a debugger?  o.O

I personally would just load it up in intellij or so, find entrance point for the client packets, set a breakpoint for that specific class that the item inserter uses to update the client, then follow it on the pause.  Excepting that I'd inject a few debugging points to the ME Terminal client-side TE on packet received to see what that all looks like and what's on the stack at the time it's called/cleared.  Not really an 'easy' task but not too hard if you know java well.

**Comment by Kobra331** [comment]
I doubt im good enough to understand the tools and the code. I'm happy to understand some code pieces of the objects in the source code. But I can try it if i have some free time.

**Comment by XenonDB** [comment]
Same problem happened to me.(V24c)
A video may better describe this situation.

[https://www.youtube.com/watch?v=PGNVMpqwGk0](https://www.youtube.com/watch?v=PGNVMpqwGk0)

**Comment by ReikaKalseki** [comment]
I knew what was meant. I still have no idea WTF is going on.
